pkg/kyc/component.go

```diff
@@ -408,6 +408,12 @@ func (c *component) validateUser(ctx context.Context, accessToken string, confRe
 	kcUser, err = c.keycloakClient.GetUser(accessToken, targetRealm, userID)
 	if err != nil {
 		c.logger.Warn(ctx, "msg", "Can't get user/accreditations", "err", err.Error())
+		switch e := err.(type) {
```

Comment:
Please add a unit test covering this scenario. 

And is there more elegant to use errors.As method ? more information https://go.dev/blog/go1.13-errors

Reply:
Other methods use the same pattern with the switch case, like [here](https://github.com/cloudtrust/keycloak-bridge/blob/master/pkg/kyc/component.go#L226) and [here](https://github.com/cloudtrust/keycloak-bridge/blob/master/pkg/support/component.go#L41), and `errors.As` are not used.

Reply:
errors.As suggested by @basbeu seems a more elegant way of testing the error type especially here where we only are interrested by one case.

Reply:
Okay thanks! I changed it.